optaplanner-docs/src/main/asciidoc/DesignPatterns/DesignPatterns-chapter.adoc

```diff
@@ -9,92 +9,74 @@
-
-. Choose the model in which *the number of planning entities is fixed during planning*.
-** For example in employee rostering, it's impossible to know in advance how many shifts each employee will have
-before Planner solves it (and it can even differ per best solution found).
+. *Annotate a many-to-one relationship with a `@PlanningEntity` annotation.* Usually the _many_ side of the relationship is the planning entity class that contains the planning variable. If the relationship is bidirectional, both sides are a planning entity class but usually the _many_ side has the planning variable and the _one_ side has the shadow variable. For example, in the Employee Rostering starter application, the `ShiftAssignment` class has an `@PlanningEntity` annotation.
+
+. *Make sure that the planning entity class has at least one problem property*. A planning entity class cannot consist of only planning variables or an ID and only planning variables.
+.. Remove any surplus `@PlanningVariable` annotations so that they become  problem properties. Doing this significantly decreases <<searchSpaceSize,the search space size>> and significantly increases solving efficiency. For example, in the Employee Rostering starter application, the `ShiftAssignment` class should not annotate both the `Shift` and `Employee` relationship with `@PlanningVariable`.
+.. Make sure that when all planning variables have a value of `null`, the planning entity instance is describable to the business people. Planning variables have a value of `null` when the planning solution is uninitialized.
+** A surrogate ID does not suffice as the required minimum of one problem property.
+** There is no need to add a hard constraint to assure that two planning entities are different. They are already different due to their problem properties.
+** In some cases, multiple planning entity instances have the same set of problem properties. In such cases, it can be useful to create an extra problem property to distinguish them. For example, in the Employee Rostering starter application, the `ShiftAssignment` class has the problem property `Shift` as well as the problem property `indexInShift` which is an `int` class.
+
+. *Choose the model in which the number of planning entities is fixed during planning*. For example, in the Employee Rostering starter application, it is impossible to know in advance how many shifts each employee will have before Optaplanner solves the model and the results can differ for each solution found.
 On the other hand, the number of employees per shift is known in advance,
-so it's better to make the `Shift` relationship a problem property
+so it is better to make the `Shift` relationship a problem property
 and the `Employee` relationship a planning variable.
 +
 image::DesignPatterns/employeeShiftRosteringModelingGuideB.png[align="center"]
 
-For inspiration, take a look at how the examples modeled their domain:
+// Not clear just what this diagram is showing:
```

Comment:
Each row is a different example and it shows the relationship in that example's data model. For example, for the "N Queens" example, the `Queen` entity has a `row` planning variable, which stores objects of type `Row`. Many `Queen`s may point to one `Row`.

Reply:
Updated.